### ISSUE TITLE
Evaluation loss is being reported, but class-based AP scores are 0?

### ISSUE THREAD

**Issue by neel04** [open-mmlab/mmdetection#6253]
Hi, I have the following log
```py
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 301/301, 3.2 task/s, elapsed: 93s, ETA:     0sLoading and preparing results...
DONE (t=0.00s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.10s).
Accumulating evaluation results...
2021-10-10 19:27:39,658 - mmdet - INFO - Evaluating bbox...
2021-10-10 19:27:40,055 - mmdet - INFO -
+-----------------+-------+-----------------+-------+---------------+-------+
| category        | AP    | category        | AP    | category      | AP    |
+-----------------+-------+-----------------+-------+---------------+-------+
| person | 0.000 | car | 0.000 | cow | 0.000 |
+-----------------+-------+-----------------+-------+---------------+-------+
2021-10-10 19:27:40,061 - mmdet - INFO - Exp name: cascade_mask_rcnn_swin_small_patch4_window7_mstrain_480-800_giou_4conv1f_adamw_3x_coco.py
2021-10-10 19:27:40,062 - mmdet - INFO - Epoch(val) [3][675] bbox_mAP: 0.0000, bbox_mAP_50: 0.0000, bbox_mAP_75: 0.0000, bbox_mAP_s: -1.0000, bbox_mAP_m: -1.0000, bbox_mAP_l: 0.0000, bbox_mAP_copypaste: 0.000 0.000 0.000 -1.000 -1.000 0.000
DONE (t=0.29s).
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = -1.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = -1.000
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = -1.000
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = -1.000
```
Any ideas why?

**Comment by TolgaOzdmir** [comment]
You may have a problem with the config file such as data train/test/val path, classes, etc.

**Comment by jshilong** [comment]
> Thanks a lot for replying so quickly guys! I can confirm my model is overfitting to the training data - and also mysteriously performing _too_ well on the validation set just because of the observed losses (validation was synthesized from train)
> 
> My dataset is pretty small, and I am using a 110M+ params model so it overfits quite quickly
> 
> ```python
> 2021-10-11 00:51:52,905 - mmdet - INFO - Epoch(val) [2][38] loss_rpn_cls: 0.0002, loss_rpn_bbox: 0.0000, s0.loss_cls: 0.0011, s0.acc: 99.9775, s0.loss_bbox: 0.0001, s1.loss_cls: 0.0010, s1.acc: 99.9634, s1.loss_bbox: 0.0001, s2.loss_cls: 0.0006, s2.acc: 99.9582, s2.loss_bbox: 0.0000, loss: 0.0032
> 
> wandb: WARNING Step must only increase in log calls.  Step 1351 < 1352; dropping 
> 
> {'val/loss_rpn_cls': 0.00016263271544702499, 'val/loss_rpn_bbox': 1.6639697366228672e-05, 'val/s0.loss_cls': 0.0011009046492721321, 'val/s0.acc': 99.97751336348684, 'val/s0.loss_bbox': 0.00014984695927092902, 'val/s1.loss_cls': 0.0010168649431235327, 'val/s1.acc': 99.96337890625, 'val/s1.loss_bbox': 7.71293784246633e-05, 'val/s2.loss_cls': 0.0006183671116320367, 'val/s2.acc': 99.95823910361842, 'val/s2.loss_bbox': 2.7828343215668083e-05, 'val/loss': 0.003170213940106087, 'learning_rate': 1.0281822390660857e-05, 'momentum': 0.9}.
> ```
> 
> From a tentative run, smaller models seem to be more robust (I will surely train them out tmrw and verify properly)
> 
> So its pretty clear that it can indeed validate @jshilong which means that the data is indeed correct (I had manually verified that the bounding boxes of images are aligned) but perhaps the annotations' _format_ might be off... thinking
> 
> I was pretty confused with the conflicting results hence the issue.
> 
> Again, many thanks for replying so quickly!
> 
> #### EDIT:-
> I am also specifying `load_from` to be a checkpoint for warm restarting - so it takes only about 50 steps to attain maximum scores. @jshilong
> ### Also a better chart
> ```python
> wandb: Run summary:
> wandb:    train/loss_rpn_cls 0.0001
> wandb:   train/loss_rpn_bbox 1e-05
> wandb:     train/s0.loss_cls 0.00046
> wandb:          train/s0.acc 99.99414
> wandb:    train/s0.loss_bbox 7e-05
> wandb:     train/s1.loss_cls 0.00039
> wandb:          train/s1.acc 99.99023
> wandb:    train/s1.loss_bbox 4e-05
> wandb:     train/s2.loss_cls 0.00023
> wandb:          train/s2.acc 99.98926
> wandb:    train/s2.loss_bbox 2e-05
> wandb:            train/loss 0.00129
> wandb:         learning_rate 1e-05
> wandb:              momentum 0.9
> wandb:              _runtime 13883
> wandb:            _timestamp 1633913442
> wandb:                 _step 1351
> wandb: Run history:
> wandb:    train/loss_rpn_cls █▅▆▃▁▂▄▁▄▂▁▃▃▃▃▁▁▂▃▂▂▁▃▂▂▂▂▂▂▃▁▂▁▁▁▁▂▁▁▁
> wandb:   train/loss_rpn_bbox ▇██▃▂▄▄▁▂▂▁▄▂▅▂▁▁▂▃▁▃▁▆▆▆▄▁▂▁▂▃▂▁▁▂▁▃▂▁▁
> wandb:     train/s0.loss_cls █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
> wandb:          train/s0.acc ▁███████████████████████████████████████
> wandb:    train/s0.loss_bbox ▄▅█▃▃▄▆▂▅▂▂▄▂▅▄▂▁▅▃▂▂▂▂▂▃▃▃▂▂▁▂▂▂▁▂▁▂▁▁▁
> wandb:     train/s1.loss_cls █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
> wandb:          train/s1.acc ▁███████████████████████████████████████
> wandb:    train/s1.loss_bbox ▄▆▇▃▄▅▇▂▆▂▄▇▄██▃▁█▄▃▂▃▃▄▅▆▄▄▄▂▄▄▃▂▃▂▃▂▁▂
> wandb:     train/s2.loss_cls █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
> wandb:          train/s2.acc ▁███████████████████████████████████████
> wandb:    train/s2.loss_bbox ▂▂▄▃▃▃▄▂▃▂▃▅▄▆█▃▁▇▄▂▃▃▄▅▆█▆▆▄▂▆▄▂▂▃▂▄▂▂▂
> wandb:            train/loss █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
> wandb:         learning_rate ████▇▇▇▇▇▇▆▆▆▆▆▆▆▅▅▅▅▅▅▄▄▄▄▄▄▃▃▃▃▃▂▂▂▁▁▁
> wandb:              _runtime ▁▁▁▂▂▂▂▂▂▃▃▃▃▃▃▄▄▄▄▄▅▅▅▅▅▅▆▆▆▆▆▇▇▇▇▇▇███
> wandb:                 _step ▁▁▁▂▂▂▂▂▂▃▃▃▃▃▄▄▄▄▄▄▅▅▅▅▅▅▆▆▆▆▆▇▇▇▇▇▇███
> ```

Sorry for the late response, I guess there are some things wrong in your test Phares, If you make sure the model's visualization is correct, the only reason is the code related to the metric or the way to load annotaions is wrong. Is there any modification of the code? you can debug by visualized your annotations and your predictions of the same image before calculating mAP, then you can localize the bug is in your annotation or in your metric calculation. Good luck!

**Comment by neel04** [comment]
Many thanks @jshilong, will try both! 🤗 

Would you happen to know if `mmdetection` has any script that easily visualizes bbox predictions @ image, otherwise I wouldn't waste time manually writing one!

I will surely update my findings here - it may be help for someone else! 🚀

**Comment by jshilong** [comment]
> Would it help If I send over my dataset for reproducibility? Because I just can't figure out this weird bug disappointed

Sorry I do not have enough time to provide such help
